contrib/test_jenkins.sh

```diff
@@ -306,6 +306,26 @@ get_rdma_device_ip_addr() {
 	echo $ipaddr
 }
 
+get_non_rdma_ip_addr() {
+    if ! which ibdev2netdev >&/dev/null
+    then
+        return
+    fi
+
+    #get the interface of the ip address that is the default gateway (pure Ethernet IPv4 address).
+    eth_iface=$(ip route show|sed -n 's/default via \(\S*\) dev \(\S*\)/\2/p'| sed -e 's/\s\+$//')
```

Comment:
why need the 2nd sed command?

Reply:
to remove the space and the end of the part before it.

```
alinas@jazz22 ~
$res=$(ip route show|sed -n 's/default via \(\S*\) dev \(\S*\)/\2/p')
alinas@jazz22 ~
$echo $res
eno1
alinas@jazz22 ~
$echo $res | wc -m
5
```

Reply:
this space was added by "echo" command

Reply:
```
++ get_non_rdma_ip_addr
++ which ibdev2netdev
+++ ip route show
+++ sed -n 's/default via \(\S*\) dev \(\S*\)/\2/p'
++ eth_iface='eno1 '
+++ ibdev2netdev
+++ grep -w eno1
++ '[' -n '' ']'
+++ ip addr show 'eno1 '
+++ awk '/inet /{print $2}'
+++ awk -F / '{print $1}'
Device "eno1 " does not exist.
++ ipaddr=
```

Reply:
pls try like this:
`sed -n 's/default via \(\S*\) dev \(\S*\).*/\2/p'`